File: DZ#GameBall/TimerAnimationProject/MenuViewController.m
//
//  MenuViewController.m
//  TimerAnimationProject
//
//  Created by Анастасия Шахлан on 14/06/2019.
//  Copyright © 2019 ASh. All rights reserved.
//

#import "MenuViewController.h"
#import "GameViewController.h"

@interface MenuViewController ()
@property (nonatomic,strong) UIButton *startGameButton;

@end

@implementation MenuViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
    
}
- (void)setupUI {
    
    self.startGameButton = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 10, 10)];
    [self.startGameButton setTitle:@"Начать игру" forState:UIControlStateNormal];
    [self.startGameButton sizeToFit];
    [self.startGameButton setTitleColor:UIColor.blueColor forState:UIControlStateNormal];
    self.startGameButton.center = CGPointMake(CGRectGetMidX(UIScreen.mainScreen.bounds), CGRectGetMidY(UIScreen.mainScreen.bounds));
    [self.startGameButton addTarget:self action:@selector(openGame) forControlEvents:UIControlEventTouchUpInside];
    
    [self.view addSubview:self.startGameButton];
}
- (void)openGame
{
    GameViewController *vc = [GameViewController new];
    [self.navigationController pushViewController:vc animated:true];
}


@end

File: DZ#CollectionViewController/DZ#CollectionViewController/MenuViewController.swift
//
//  ViewController.swift
//  DZ#CollectionViewController
//
//  Created by Анастасия Шахлан on 02/06/2019.
//  Copyright © 2019 ASh. All rights reserved.
//

import UIKit

class MenuViewController: UIViewController {
    
    var buttonLightLevel: UIButton!
    var buttonHardLevel : UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        prepareUI()
        
    }
    
    func prepareUI() {

        self.view.backgroundColor = .white
        
        buttonLightLevel = UIButton(type: .system)
        buttonLightLevel.setTitle("Легкий уровень", for: .normal)
        buttonLightLevel.sizeToFit()
        buttonLightLevel.center = CGPoint(x: view.center.x, y: view.center.y - 30)
        buttonLightLevel.accessibilityIdentifier = "buttonLightLevel"
        buttonLightLevel.addTarget(self, action: #selector(openGame(sender:)), for: .touchUpInside)
        view.addSubview(buttonLightLevel)
        
        buttonHardLevel = UIButton(type: .system)
        buttonHardLevel.setTitle("Сложный уровень", for: .normal)
        buttonHardLevel.sizeToFit()
        buttonHardLevel.center = CGPoint(x: view.center.x, y: view.center.y + 30)
        buttonHardLevel.accessibilityIdentifier = "buttonHardLevel"
        buttonHardLevel.addTarget(self, action: #selector(openGame(sender:)), for: .touchUpInside)
        view.addSubview(buttonHardLevel)
        
    }
    
    @objc
    func openGame(sender: UIButton) {
        let vc = GameViewController()
        
        if sender.accessibilityIdentifier == "buttonLightLevel" {
            vc.level = .light
        } else {
            vc.level = .hard
        }
        
        self.navigationController?.pushViewController(vc, animated: true)
        
    }

}

